bounties_api/analytics/management/commands/timeline_generator.py

```diff
@@ -260,10 +267,41 @@ def handle(self, *args, **options):
                 # and provides a better way to expose by day or by hour in case of been needed
                 for day in days:
                     bounty_day = generate_timeline(day_bounds(day), schema=schema)
-                    bounty_points = BountiesTimeline.objects.filter(date=day.date(), schema=schema)
+                    bounty_points = BountiesTimeline.objects.filter(date=day.date(), schema=schema, is_week=False)
 
                     if bounty_points.exists():
                         bounty_point = bounty_points.first()
                         bounty_day.id = bounty_point.id
 
                     bounty_day.save()
+
+        for schema in schemas:
+            needs_genesis = not BountiesTimeline.objects.filter(schema=schema, is_week=True).exists()
+
+            first_date = BountyState.objects.first()
+            last_date = BountyState.objects.last()
+
+            if needs_genesis:
+                bounties_by_week = range_weeks(first_date.change_date, last_date.change_date + timedelta(days=7))
+                
+                for week in bounties_by_week:
+                    bounty_week = generate_timeline(week_bounds(week), schema=schema)
+                    bounty_week.is_week = True
+
+                    bounty_week.save()
+            else:
+                last_update = BountiesTimeline.objects.filter(schema=schema, is_week=True).order_by('date').last()
+                since = arrow.get(last_update.date).to('utc')
+                weeks = range_weeks(since, datetime.utcnow())
+
+                for week in weeks:
+                    bounds = week_bounds(week)
+                    bounty_week = generate_timeline(bounds, schema=schema)
+                    bounty_points = BountiesTimeline.objects.filter(date=bounds[0].date(), schema=schema, is_week=True)
```

Comment:
The date that is saved should be the end-bound date if that makes sense, not the start date for the week.